### PR TITLE
Added object lock testcase

### PR DESCRIPTION
**Pull request by vivekanandan-k-rh**
Updated in polarion ticket: https://polarion.engineering.redhat.com/polarion/#/project/CEPH/workitem?id=CEPH-83574055

**Test steps:**
Create bucket with bucket lock enabled
Enable object lock configuration
Upload object to the bucket
Upload another object in same bucket
Verify new version is created for object
Verify deletion of object versions are disabled

**Automation PR:**

https://github.com/red-hat-storage/ceph-qe-scripts/pull/154

### CODE REVIEW

**Comment by rakeshgm** [comment]
the test case is not in approved state. the checklist had all the items, it makes sense to check off when you are raising  the PR instead of deleting the template.

**Comment by vivekanandan-k-rh** [comment]
cephci log: http://pastebin.test.redhat.com/982573

File: suites/pacific/rgw/tier_1_object.yaml [hunk 1]
```diff
@@ -124,4 +124,13 @@ tests:
         script-name: test_swift_basic_ops.py
         config-file-name: test_swift_object_expire_op.yaml
         timeout: 500
-        
+   - test:
+      name: object lock verification
+      desc: object lock test in s3
+      polarion-id: CEPH-83574055
+      module: sanity_rgw.py
+      config:
```
Comment (psathyan) [inline suites/pacific/rgw/tier_1_object.yaml:132]: I think you need to give an additional space...
```
- test:
    name: object lock verification
    desc: ...
```

Comment (vivekanandan-k-rh) [reply suites/pacific/rgw/tier_1_object.yaml]: Modified it accordingly to align with existing yaml.

**Comment by vivekanandan-k-rh** [comment]
> the test case is not in approved state in polarion. the checklist had all the items, it makes sense to check off when you are raising the PR instead of deleting the template.

Sure. Will follow it in further commits

**Review by rakeshgm — APPROVED** [review]

**Pull request merged**